packages/dashboard/src/components/internalDashboard/collapsiblePanel.tsx

```diff
@@ -1,19 +1,95 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
 
 import './index.css';
-import { colorBackgroundButtonPrimaryDefault, spaceContainerHorizontal } from '@cloudscape-design/design-tokens';
+import {
+  colorBackgroundButtonPrimaryDefault,
+  colorBorderDividerDefault,
+  spaceContainerHorizontal,
+  spaceStaticXxxs,
+} from '@cloudscape-design/design-tokens';
+import Box, { BoxProps } from '@cloudscape-design/components/box';
+import Button from '@cloudscape-design/components/button';
+import { IconProps } from '@cloudscape-design/components/icon';
+import { DEFAULT_COLLAPSED_SIDE_PANE_WIDTH } from '../resizablePanes/constants';
 
-export function CollapsiblePanel(props: { icon: string; dataCy: string }) {
-  return (
+type CollapsiblePanelProps = {
+  isPanelCollapsed: boolean;
```

Comment:
is it possible to track the collapsed state in this component?

Reply:
What do you mean? Like have this component manage the panel collapsed state? 
The resizablePanes component manages the left and right collapsible panels. It needs to manage this state for its own styling, which is then passed to the collapsible panel components.

Reply:
Oh, right, I forgot that special styling exists in the root component.